### PR TITLE
[DO NOT MERGE - WIP] Cirrus: use Ubuntu 22.04 LTS

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

#### Does this PR introduce a user-facing change?

```release-note
None
```

depends on https://github.com/containers/automation_images/pull/134

### CODE REVIEW

**Comment by cevich** [comment]
And now [the "real fun" begins (log).](https://cirrus-ci.com/build/5974831744155648) :grin:

**Comment by lsm5** [comment]
@mtrmac ping, any idea why the `too many collisions` in this PR? See : https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6348867829497856/html/int-podman-ubuntu-2204-root-host.log.html#t--Do-not-build-image-at-all-if-build-false--3

**Comment by mtrmac** [comment]
(The linked log seems to be a straightforward repo unavailability; looking at https://api.cirrus-ci.com/v1/task/5492536713150464/logs/main.log instead.)

- Unlikely hypothesis:
  - The test is supposed to generate random `lockPath` names in `PodmanTestCreateUtil`; maybe the RNG is being set up by Ginkgo to always generate the same names. That could be verified by adding logging of the `lockPath` value. (If the generated names are truly random, I have absolutely no idea what’s going on. If they aren’t random, that’s not a _problem_ per se, just a precondition for the hypothesis.)
  - The design requires each test to cleanup up the remote socket after itself when done, so at least assuming there are not 1000 tests running in parallel, an available path should exist even if the names are deterministic. Typically that seems to happen via `podmanTest.Cleanup()` → `StopRemoteService`. Is that cleanup missing in some situations?
  - Even assuming the cleanup is missing or not performed, it is seriously surprising to me that the test log shows a failure on just the 16th test, when there are supposed to be 1000 attempts. Is the test somehow running on a pre-existing directory with close to 1000 lock files already? (And is the RNG seed consistent across runs?)
- Alternative hypothesis: The `lockFile` creation is failing for some other, consistent, reason, maybe a permission error; so a single attempt to create the lock file runs out of all 1000 attempts. Logging the errors returned by `os.OpenFile(lockPath)` would reveal that. Guessing, maybe the current directory (I don’t even know what that directory is) is not writable.

**Review by cevich — APPROVED** [review]
Changes LGTM

**Comment by cevich** [comment]
The RNG/name-collision problem was seen in Fedora-land last year I believe.  Miloslav and I (mostly Miloslav) spent quite a bit of time poking at it.  I would point out that we do have [a software RNG service (rngd) enabled in the Fedora images](https://github.com/containers/automation_images/blob/main/base_images/fedora_base-setup.sh#L24-L44), but not explicitly enabled in Ubuntu (IIRC).  If there isn't one by default, it's possible this needs to be installed/enabled for Ubuntu.

**Comment by mtrmac** [comment]
Yes, that’s what the “alternative hypothesis” part of https://github.com/containers/podman/pull/14397#issuecomment-1141486843 suggests. Do y’all need me to prepare a patch to that effect?

**Comment by lsm5** [comment]
> Yes, that’s what the “alternative hypothesis” part of [#14397 (comment)](https://github.com/containers/podman/pull/14397#issuecomment-1141486843) suggests. Do y’all need me to prepare a patch to that effect?

na, looking into it now. Let me get back to you.

**Comment by cevich** [comment]
@lsm5 as per https://paste.centos.org/view/f7b8a0f8 I'm onboard with [mtrmac](https://github.com/mtrmac).  Damn lock file simply isn't being created for some reason specific to Ubuntu :confused: 

I peeked at the other failures here and TBH I'd just ignore them until the lockfile thing is figured out.  Hopefully by then there won't be too many more new problems (and maybe less if we're luck) :grin:

**Comment by lsm5** [comment]
> @lsm5 looking better now. The next-worse problem seems to be the ` [Fail] Podman manifest [It] authenticated push` in both Fedora and Ubuntu.

rerunning them now. Could it be because registry was down or something?

**Comment by lsm5** [comment]
```
[+0673s] Error: pod 11cad2f5b91a006055ed097c6527926a5e061b4829380627e6443a79b4659be2 not found in database: no such pod
[+0673s] Error: pod 02c775b342c144cd0d780e6cec8c160f466fd633f0dc7199c435432891136f96 not found in database: no such pod
[+0673s] Error: pod 267e1676159880afb3444c97398a1f6cbb3d04e5b23d56955037c3afd21446cc not found in database: no such pod
```

**Comment by cevich** [comment]
> Could it be because registry was down or something?

IIRC this is a locally run registry container.  But yeah, it could be a flake, worth checking open-issues if it reproduces.

**Comment by cevich** [comment]
> not found in database: no such pod

Yeah, that's concerning.

**Comment by cevich** [comment]
force-push: Rebased on main.

**Comment by lsm5** [comment]
> Interesting debugging data-point: I used `hack/get_ci_vm.sh int podman ubuntu-2204 root host` and edited `test/e2e/manifest_test.go:296` to use `FIt()` (focus-mode) and the `authenticated push` test passed on the first (and only) run! So perhaps some other test is screwing something up on the system that causes it to fail (and also on Ubuntu remote).

I tried this on 5 tests together along with the authenticated push one, and all seemed to pass. It's only when *all* tests are run that this seems to fail. Might be better to get some other eyes on this.

**Comment by cevich** [comment]
@containers/podman-maintainers PTAL.  We're having a major problem with the new Ubuntu and updated Fedora images.  The longer this PR remains un-merged, the higher the changes additional breaking issues will make it into `main` :disappointed: Here's a summary of what we need help with:

* `int podman fedora-36 root container` - [**Many** tests are failing with a 90-second timeout error (log)](https://cirrus-ci.com/task/6238864456024064?logs=main#L2249).  This is causing the entire cirrus-task to timeout, meaning we loose a lot of  other useful state/logging information (including the annotated logs).  Something new in this Fedora image must be fundamentally broken for podman-in-podman WRT current podman code.  This is bad and reproducible.
* `int remote fedora-36 root host` - [The `Podman build [It] Remote build .containerignore filtering embedded directory` test fails (log)](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4550014595760128/html/int-remote-fedora-36-root-host.log.html), also a timeout of some sort.  This is reproducible.
* `int podman ubuntu-2204 root host` - [The `Podman manifest [It] authenticated push` test fails (log)](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6696261293178880/html/int-podman-ubuntu-2204-root-host.log.html#t--authenticated-push--1).  This same registry image is used on the old Ubuntu 2110, Fedora, and new Fedora images for integration testing, without this failure.  Clearly something is borkt either with the image or the new Ubuntu environment, we just have no clue what it could be.  This problem is reproducible, but only if running all tests.  Perhaps some other test isn't cleaning up properly?

**Comment by Luap99** [comment]
re: 90 sec timeout in container test

It looks like all timeouts are in the podman checkpoints tests, so I suspect an issue with that.

**Comment by cevich** [comment]
> It looks like all timeouts are in the podman checkpoints tests, so I suspect an issue with that.

Oh! Durr, I feel really dumb not noticing that :relaxed: 

@adrianreber Remind me, I seem to remember that the `checkpoint` tests are invalid for a podman-in-podman test environment.   If they are, then do you have any idea why [they might be failing (raw log)](https://api.cirrus-ci.com/v1/task/6238864456024064/logs/main.log) in the latest Fedora (with updates-testing repo.)?

**Comment by cevich** [comment]
> That is weird, most likely the loopback interface is down, check ip addr show dev lo

I should think that would show up by other tests failing then, no?  Reminder: It only breaks when we run all the tests, if we isolate `authenticated push` it passes no problem :confused:

**Comment by cevich** [comment]
Anyone have a guess on:  `int remote fedora-36 root host` - [The Podman build [It] Remote build .containerignore filtering embedded directory test fails (log)](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4550014595760128/html/int-remote-fedora-36-root-host.log.html), also a timeout of some sort. This is reproducible.

**Comment by lsm5** [comment]
@cevich If I use the same registry startup code for `authenticated push` as the one in login_logout_test.go, the authenticated push will consistently pass. I haven't yet been able to narrow things down further.

@vrothberg wdyt? should/can the registry startups be unified? Or is there good reason behind them being separate?

**Comment by edsantiago** [comment]
Well, your [f35 failure](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5093701889294336/html/int-remote-fedora-35-root-host.log.html) is the unpleasant new #14653 flake. f36 remote looks like a flake too, I've restarted it.

[ubuntu local](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4530751935873024/html/int-podman-ubuntu-2204-root-host.log.html#t--authenticated-push--1) looks like a networking problem:
```
         # podman [options] push --tls-verify=false --creds=userZqcY:XdNYDe1NtY9jbZ0 --format=v2s2 localhost:5894/alpine:latest
         Getting image source signatures
         time="2022-06-17T14:18:13Z" level=warning msg="Failed, retrying in 1s ... (1/3). Error: trying to reuse blob sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0 at destination: pinging container registry localhost:5894: Get \"http://localhost:5894/v2/\": dial tcp 127.0.0.1:5894: connect: no route to host"
         Getting image source signatures
         time="2022-06-17T14:18:26Z" level=warning msg="Failed, retrying in 1s ... (2/3). Error: trying to reuse blob sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0 at destination: pinging container registry localhost:5894: Get \"http://localhost:5894/v2/\": dial tcp 127.0.0.1:5894: connect: no route to host"
         Getting image source signatures
         time="2022-06-17T14:18:38Z" level=warning msg="Failed, retrying in 1s ... (3/3). Error: trying to reuse blob sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0 at destination: pinging container registry localhost:5894: Get \"http://localhost:5894/v2/\": dial tcp 127.0.0.1:5894: connect: no route to host"
         Getting image source signatures
         Error: trying to reuse blob sha256:03901b4a2ea88eeaad62dbe59b072b28b6efa00491962b8741081c5df50c65e0 at destination: pinging container registry localhost:5894: Get "http://localhost:5894/v2/": dial tcp 127.0.0.1:5894: connect: no route to host
```
What's weird is, if localhost networking were broken, I would expect a LOT more tests to fail. Nope, it's only this one test. And it's not a port conflict, because three retries get three different ports. @Luap99 any suggestions on how to find out more info in this case?

**Comment by Luap99** [comment]
> What's weird is, if localhost networking were broken, I would expect a LOT more tests to fail. Nope, it's only this one test. And it's not a port conflict, because three retries get three different ports. @Luap99 any suggestions on how to find out more info in this case?

see https://github.com/containers/podman/pull/14397#issuecomment-1155587234

`no route to host` is a very weird error, I guess if this is a container with forwarded ports this can happen when the container ip is not reachable. Why this happens no idea, most likely some conflict/cleanup error. Can you get the `iptables -nvL -t nat` output when this happens?

**Comment by edsantiago** [comment]
>the ubuntu failure looks like a registry conflict

@lsm5 I don't understand what that means. No such conflict should be possible. I think @Luap99 's suggestion is worth pursuing: can you instrument the test to run (and display the output of) the `iptables` command he suggests? Running it immediately before the failing podman commands (which seem consistent) is much easier than running it on failure.

**Comment by cevich** [comment]
@lsm5 slightly newer images: `c4514861978746880` you could try (from #14719), but I'm not hopeful there will be much difference.  I tried to make it clear in my PR that we need to get this one fixed & merged first.  What problems are outstanding here, that you're not currently working on (so I can help with them in parallel)?

**Comment by lsm5** [comment]
> @lsm5 slightly newer images: `c4514861978746880` you could try (from #14719), but I'm not hopeful there will be much difference. I tried to make it clear in my PR that we need to get this one fixed & merged first. What problems are outstanding here, that you're not currently working on (so I can help with them in parallel)?

@cevich pushed a change with the new id. RE: outstanding problems, the same ones as before I guess. `authenticated push` and the `90s` timeout. Last week I was occupied with RHEL and I'm being asked to spend more time on arm CI. So, if anyone else wants to take this over, please feel free.

File: test/e2e/common_test.go [hunk 1]
```diff
@@ -296,6 +296,13 @@ func PodmanTestCreateUtil(tempDir string, remote bool) *PodmanTestIntegration {
 		var pathPrefix string
 		if !rootless.IsRootless() {
 			pathPrefix = "/run/podman/podman"
+			// FIXME: podman system service fails to start on Ubuntu 22.04 for
+			// some reason because of which we're creating /run/podman if it
+			// doesn't exist.
+			if _, err := os.Stat("/run/podman"); os.IsNotExist(err) {
+				mkdirErr := os.MkdirAll("/run/podman", 0755)
```
Comment (edsantiago) [inline test/e2e/common_test.go:303]: Suggestion: `0700`, not 755. And @containers/podman-maintainers I think a better solution would be to understand why Ubuntu is doing this.

Comment (Luap99) [reply test/e2e/common_test.go:303]: Is the podman tmpfiles.d file installed correctly?
```
$ grep /run/podman /usr/lib/tmpfiles.d/podman.conf
D! /run/podman 0700 root root

```

**Comment by cevich** [comment]
> Last week I was occupied with RHEL and I'm being asked to spend more time on arm CI. So, if anyone else wants to take this over, please feel free.

I hear ya...these image-update PRs are a major time-sink and always need lots of hand-holding and teamwork.  Speaking of which, I s'pose there's no reason other maintainers can't just force-push to your branch to try and kick this PR down the road.

File: test/e2e/common_test.go (hunk 1 above)
Comment (cevich) [reply test/e2e/common_test.go:303]: For podman CI, we remove all the files from the RPM then are suppose to do a `make install`.  After looking at the `Makefile`, we should be installing.   Checking if the logs give any clues why it's not there...

Comment (cevich) [reply test/e2e/common_test.go:303]: Oh! So when I looked at the VM, I canceled the setup process - thinking (wrongly) this file was installed at VM image build time.  Checking again, the file is definitely there.  Maybe is there some systemd service that needs to be restarted/reloaded to pick up the file?  If so, I wonder how is/was this never a problem before :man_shrugging:

Comment (cevich) [reply test/e2e/common_test.go:303]: I believe I follow what's happening here.  The entry in `podman.conf` is `D! /run/podman 0700 root root`, which means "Only create this directory on boot".  So the `make install`, nor any amount of service reloading will create this directory at test runtime.  Though, there should have been a podman package installed on the VM during the build process.  **THAT** should have placed `podman.conf` file such that the directory gets created on boot.  So maybe the Ubuntu podman package is broken then...

Comment (lsm5) [reply test/e2e/common_test.go:303]: @cevich fixed in the latest build. For whatever reason, debbuild can't seem to make sense of `%{_tmpfilesdir}` so I explicitly spelled out the path.

Comment (cevich) [reply test/e2e/common_test.go]: Great, I still have an image build PR open, I'll trigger a re-build there...

Comment (cevich) [reply test/e2e/common_test.go]: ...[Images building here.](https://github.com/containers/automation_images/pull/140)

Comment (cevich) [reply test/e2e/common_test.go]: @lsm5 I confirmed in a newly build Ubuntu container image that the `/usr/lib/tmpfiles.d/podman.conf` is there now.  Thanks :D

**Comment by cevich** [comment]
Update: `Podman build` (remote) only failed on my #14719 but not here, so that must be a flake.  Meaning only `authenticated push` needs debugging/fixing :birthday:

**Comment by edsantiago** [comment]
I think there's something broken in networking. I did a `re-run with console`, and while tests were running:
```console
# hack/podman-registry start
PODMAN_REGISTRY_IMAGE="quay.io/libpod/registry:2.6"
PODMAN_REGISTRY_PORT="5055"
PODMAN_REGISTRY_USER="user0cDX"
PODMAN_REGISTRY_PASS="keyT0ZSieGrqCy3"
# telnet localhost 5055
Trying 127.0.0.1...
telnet: Unable to connect to remote host: No route to host
```
`lsof` on the process shows it listening:
```
conmon  178016 root    5u     IPv4             706971      0t0     TCP *:5055 (LISTEN)
```
There is a bad interface on the system:
```console
# ip a
...
3: cni-podman0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 12:d0:4c:e4:e6:c3 brd ff:ff:ff:ff:ff:ff
    inet 10.88.0.1/16 brd 10.88.255.255 scope global cni-podman0
       valid_lft forever preferred_lft forever
    inet6 fe80::10d0:4cff:fee4:e6c3/64 scope link 
       valid_lft forever preferred_lft forever
```
Deleting it (via `ip link del`) makes the port accessible again. So, something seems to be creating the old-style CNI interface somehow. But what?

@Luap99 here's the iptables output you requested:
```console
# iptables -nvL -t nat
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
   19  9639 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
    1    32 NETAVARK-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
 1654  107K CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL
  133  8891 NETAVARK-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
  488 30887 NETAVARK-HOSTPORT-MASQ  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 3042  192K CNI-HOSTPORT-MASQ  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd requiring masquerade */
    0     0 CNI-a8255fdfc238d8dd191b91e2  all  --  *      *       10.88.0.176          0.0.0.0/0            /* name: "podman" id: "0efdabc90252e4ecb9d07ac8dfb8dd0289c6fbb8b4420e1efcf7a3688913690a" */
    0     0 CNI-e060fd9781bbc14435e13cb1  all  --  *      *       10.88.2.67           0.0.0.0/0            /* name: "podman" id: "8d20cf97764f17ec4fe5bc64ab301a043803ea175260465085d0defa9685c20e" */
    0     0 CNI-d2034e09d0207d90cb0b3c73  all  --  *      *       10.88.73.146         0.0.0.0/0            /* name: "podman" id: "0af3ee4d0991539a49ffbd4bfd1e7b512a5a3e981ed63880c486a60f0230107c" */
    0     0 CNI-e9768ba8e3d20a3ac43ebc4d  all  --  *      *       10.88.73.147         0.0.0.0/0            /* name: "podman" id: "82f1c45e16095ed236ada94be140f8dcd82ea0d6e92fadc82412bb732e08b369" */
   13   828 NETAVARK-1D8721804F16F  all  --  *      *       10.88.0.0/16         0.0.0.0/0           
    0     0 CNI-0b1cafeb9628ee0b55490e13  all  --  *      *       10.88.0.16           0.0.0.0/0            /* name: "podman" id: "edfe69e088678a6d08c47f634caf413dc6b874b33e644d2b20cf6559d43a3a58" */
    0     0 CNI-8b675d9317655824ae30fa13  all  --  *      *       10.88.35.217         0.0.0.0/0            /* name: "podman" id: "1969dd24f05c18fb26aa82daa25045dff36879132efb2d495cf2992d80c70b4f" */
    0     0 CNI-21bee5fab41f61342381d410  all  --  *      *       10.88.35.232         0.0.0.0/0            /* name: "podman" id: "14f96381d6958166bea5ef1d81c70570b07ff4505c30894f19afc0d38b82592d" */
    0     0 CNI-4e1cb35a555b72d297edfd63  all  --  *      *       10.88.35.233         0.0.0.0/0            /* name: "podman" id: "469afa180258626ce228bd42c83fb53b10d34ec5e91b72d9bce3542072e3463d" */

Chain CNI-0b1cafeb9628ee0b55490e13 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "edfe69e088678a6d08c47f634caf413dc6b874b33e644d2b20cf6559d43a3a58" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "edfe69e088678a6d08c47f634caf413dc6b874b33e644d2b20cf6559d43a3a58" */

Chain CNI-21bee5fab41f61342381d410 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "14f96381d6958166bea5ef1d81c70570b07ff4505c30894f19afc0d38b82592d" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "14f96381d6958166bea5ef1d81c70570b07ff4505c30894f19afc0d38b82592d" */

Chain CNI-4e1cb35a555b72d297edfd63 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "469afa180258626ce228bd42c83fb53b10d34ec5e91b72d9bce3542072e3463d" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "469afa180258626ce228bd42c83fb53b10d34ec5e91b72d9bce3542072e3463d" */

Chain CNI-8b675d9317655824ae30fa13 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "1969dd24f05c18fb26aa82daa25045dff36879132efb2d495cf2992d80c70b4f" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "1969dd24f05c18fb26aa82daa25045dff36879132efb2d495cf2992d80c70b4f" */

Chain CNI-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination         

Chain CNI-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination         
  119  7140 MASQUERADE  all  --  *      *       0.0.0.0/0            0.0.0.0/0            mark match 0x2000/0x2000

Chain CNI-HOSTPORT-SETMARK (0 references)
 pkts bytes target     prot opt in     out     source               destination         
  296 17760 MARK       all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* CNI portfwd masquerade mark */ MARK or 0x2000

Chain CNI-a8255fdfc238d8dd191b91e2 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "0efdabc90252e4ecb9d07ac8dfb8dd0289c6fbb8b4420e1efcf7a3688913690a" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "0efdabc90252e4ecb9d07ac8dfb8dd0289c6fbb8b4420e1efcf7a3688913690a" */

Chain CNI-d2034e09d0207d90cb0b3c73 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "0af3ee4d0991539a49ffbd4bfd1e7b512a5a3e981ed63880c486a60f0230107c" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "0af3ee4d0991539a49ffbd4bfd1e7b512a5a3e981ed63880c486a60f0230107c" */

Chain CNI-e060fd9781bbc14435e13cb1 (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "8d20cf97764f17ec4fe5bc64ab301a043803ea175260465085d0defa9685c20e" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "8d20cf97764f17ec4fe5bc64ab301a043803ea175260465085d0defa9685c20e" */

Chain CNI-e9768ba8e3d20a3ac43ebc4d (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16         /* name: "podman" id: "82f1c45e16095ed236ada94be140f8dcd82ea0d6e92fadc82412bb732e08b369" */
    0     0 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4          /* name: "podman" id: "82f1c45e16095ed236ada94be140f8dcd82ea0d6e92fadc82412bb732e08b369" */

Chain NETAVARK-1D8721804F16F (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    1    60 ACCEPT     all  --  *      *       0.0.0.0/0            10.88.0.0/16        
   12   768 MASQUERADE  all  --  *      *       0.0.0.0/0           !224.0.0.0/4         

Chain NETAVARK-DN-1D8721804F16F (1 references)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 NETAVARK-HOSTPORT-SETMARK  tcp  --  *      *       10.88.0.0/16         0.0.0.0/0            tcp dpt:5055
    3   180 NETAVARK-HOSTPORT-SETMARK  tcp  --  *      *       127.0.0.1            0.0.0.0/0            tcp dpt:5055
    3   180 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:5055 to:10.88.0.3:5000

Chain NETAVARK-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    3   180 NETAVARK-DN-1D8721804F16F  tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:5055 /* dnat name: podman id: 5a0f24868116ef2bd39df01e9065b1a8775584491afd2cd34cf72bcaa0134864 */

Chain NETAVARK-HOSTPORT-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination         
  180 10800 MASQUERADE  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* netavark portfw masq mark */ mark match 0x2000/0x2000

Chain NETAVARK-HOSTPORT-SETMARK (2 references)
 pkts bytes target     prot opt in     out     source               destination         
    3   180 MARK       all  --  *      *       0.0.0.0/0            0.0.0.0/0            MARK or 0x2000
```

**Comment by edsantiago** [comment]
>Or they do not use the correct --network-backend option that is set by the e2e tests.

Thanks you, that's it.

`manifest_test.go` is the only test that uses `hack/podman-registry-go`:
https://github.com/containers/podman/blob/cf747399b13432d000cfd9556a248681363dda2d/test/e2e/manifest_test.go#L297-L300

(side note: the double use of `registry` here is horrible: it's both a constant and a variable. Reminder to self, fix that).

Anyhow, `hack/podman-registry-go` invokes `hack/podman-registry`, which runs `bin/podman` but has no way of knowing the magic options used in e2e tests.

SOLUTION: get rid of the `hack/podman-registry-go` stuff. Just run the registry however all the other tests do it.

**Comment by cevich** [comment]
> because that way lies madness.

Lol, thanks for checking into this Ed and Paul.  FYI: I'm pretty sure during setup we `make install` so the system podman ***should*** be the same as `./bin/podman`.  However, I cannot predict what evil lurks in `/etc/containers/*` or the runtime database, so it's easily possible something causes it to revert to CNI.

**Comment by edsantiago** [comment]
>during setup we make install

That doesn't matter. The short version of the problem is: e2e tests make excessive use of podman command line args (`podman --this --that --oh --my --gosh --so --many --options), hence, anything that invokes plain `podman` (like the registry script) is gonna screw up the system.

But it's moot, because my fixer PR is merged, so, @lsm5, please rebase & repush, this should pass now.

**Comment by edsantiago** [comment]
>Well screw up any future e2e runs

No, I mean, screw up the e2e test run right smack in the middle of the run. It's kind of complicated, but here's the skeleton:
* e2e tests run, blah blah, yeah, going fine
* one of the e2e tests invokes a script that runs plain `podman`, not `podman --with --all --the --options`
* ker-blammo. Everything gets destroyed because plain podman switches the system from netavark to CNI and all goes to hell

Does that make sense?

**Comment by cevich** [comment]
> Does that make sense?

Oh I see, gotcha, okay thanks.

**Comment by edsantiago** [comment]
Aw, phooey, @cevich, can I ask you to reconsider? All this needs is a rebase & push, and life will (should) be happy. We can get Ubuntu 22 right now, all we need is this PR to merge. The criu issue does not affect this PR.

**Pull request closed without merging**